crates/noirc_evaluator/src/ssa_refactor/opt/constant_folding.rs

```diff
@@ -19,7 +22,13 @@ impl Ssa {
     /// now be simplified.
     pub(crate) fn fold_constants(mut self) -> Ssa {
         for function in self.functions.values_mut() {
-            constant_fold(function);
+            match function.runtime() {
+                RuntimeType::Acir => {
+                    constant_fold(function);
+                }
+                // Brillig is already generated at this step, so we cannot operate on brillig functions anymore.
+                RuntimeType::Brillig => {}
```

Comment:
I think this step was added because of the bug that was fixed in #1611. Lets merge that PR then we can remove this snippet